pkg/types/builder.go

```diff
@@ -156,11 +156,14 @@ func (g *Builder) buildResource(res *schema.Resource, cfg *config.Resource, tfPa
 			// If it is an observation field, it will be dropped.
 			// Data will be loaded from the referenced secret key.
 			fieldNameCamel += sfx
-			// todo(hasan): do we need the pointer type if optional?
-			fieldType = typeSecretKeySelector
 
-			jsonTag = fmt.Sprintf("%s,omitempty", NewNameFromCamel(fieldNameCamel).LowerCamelComputed)
 			tfTag = "-"
+			fieldType = typeSecretKeySelector
+			jsonTag = fmt.Sprintf("%s", NewNameFromCamel(fieldNameCamel).LowerCamelComputed)
+			if sch.Optional {
```

Comment:
Don't we need to skip [here](https://github.com/crossplane-contrib/terrajet/blob/6d7fb7e/pkg/resource/sensitive.go#L167) if the value of that pointer is `nil`?

Reply:
No, because expandWildcards method [here](https://github.com/crossplane-contrib/terrajet/blob/6d7fb7e/pkg/resource/sensitive.go#L162), does only return paths that are available in the spec.